community/modules/compute/schedmd-slurm-gcp-v5-node-group/source_image_logic.tf

```diff
@@ -16,52 +16,46 @@
   }
 
-  source_image         = lookup(var.instance_image, "name", "")
-  source_image_family  = lookup(var.instance_image, "family", "")
-  source_image_project = lookup(var.instance_image, "project", "")
-  source_image_project_normalized = (
-    local.source_image != "" || length(regexall("/", local.source_image_project)) > 0
-    ? local.source_image_project
-    : "projects/${local.source_image_project}/global/images/family"
+  # this approach to "hacking" the project name allows a chain of Terraform
+  # calls to set the instance source_image (boot disk) with a "relative resource
+  # name" that passes muster with VPC Service Control rules
+  #
+  # https://github.com/terraform-google-modules/terraform-google-vm/blob/735bd415fc5f034d46aa0de7922e8fada2327c0c/modules/instance_template/main.tf#L28
+  # https://cloud.google.com/apis/design/resource_names#relative_resource_name
+  source_image_project_normalized = (can(var.instance_image.family) ?
+    "projects/${data.google_compute_image.slurm.project}/global/images/family" :
+    "projects/${data.google_compute_image.slurm.project}/global/images"
   )
+  source_image_family = can(var.instance_image.family) ? data.google_compute_image.slurm.family : ""
+  source_image        = can(var.instance_image.name) ? data.google_compute_image.slurm.name : ""
 }
 
 data "google_compute_image" "slurm" {
-  family  = contains(keys(var.instance_image), "family") ? var.instance_image["family"] : null
-  name    = contains(keys(var.instance_image), "name") ? var.instance_image["name"] : null
-  project = local.source_image_project
+  family  = try(var.instance_image.family, null)
+  name    = try(var.instance_image.name, null)
+  project = var.instance_image.project
 
   lifecycle {
     postcondition {
-      condition = !var.instance_image_compatible || try(contains(local.supported_project_families[self.project], self.family), false)
-      error_message = (contains(keys(local.supported_project_families), self.project) ? <<-EOD
-      The image specified in the blueprint was found, but it is not compatible with this HPC Toolkit for Slurm module. The compatible image families are:
-
-      ${format("%-30s%-30s", "PROJECT", "IMAGE FAMILY")}
-      ${join("\n", [for proj in keys(local.supported_project_families) : join("\n", [for family in local.supported_project_families[proj] : format("%-30s%-30s", proj, family)])])}
-
-      Images that are not within this family may not have Slurm installed by default or may have faulty configurations.  If you are attempting to deploy a custom Slurm image, or an image with an unexpected version, please set the "instance_image_compatible" variable to "false" in the blueprint.
-      See https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/docs/vm-images.md#slurm-on-gcp for more information.
+      condition     = var.instance_image_custom || try(contains(keys(local.known_project_families), self.project), false)
```

Comment:
you do not need to wrap this in a `try` function.

Reply:
Resolved.